src/sonic-bgpcfgd/bgpcfgd/managers_device_global.py

```diff
@@ -142,6 +150,54 @@ def configure_idf(self, data=None):
         else:
             log_notice("DeviceGlobalCfgMgr:: IDF configuration is up-to-date")
 
+    def configure_received_bandwidth(self, data=None):
+        """Configure received bandwidth for W-ECMP feature"""
+        
+        state = self.RECEIVED_DEFAULTS
+        
+        if data is not None:
+            if "received_bandwidth" in data:
+                state = data["received_bandwidth"]
+        
+        if self.is_update_required("received_bandwidth", state):
+            if self.set_received_bandwidth(state):
+                self.directory.put(self.db_name, self.table_name, "received_bandwidth", state)
+        else:
+            log_notice("DeviceGlobalCfgMgr:: received bandwidth for W-ECMP configuration is up-to-date")
+     
+    def set_received_bandwidth(self, status):
+        """API to configure received bandwidth for W-ECMP"""
+        
+        bgp_asn = self.directory.get_slot("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME)["localhost"]["bgp_asn"]
```

Comment:
@colintle are we sure that BGP ASN will be there at the time we decide to add a new configuration. What if missing?

Reply:
src/sonic-bgpcfgd/bgpcfgd/managers_advertise_rt.py also does this too